lib/core/ppp_service/ppp.dart

```diff
@@ -0,0 +1,136 @@
+import 'package:amplify_api/amplify_api.dart';
+import 'package:amplify_auth_cognito/amplify_auth_cognito.dart';
+import 'package:amplify_flutter/amplify_flutter.dart';
+import 'package:amplify_storage_s3/amplify_storage_s3.dart';
+import 'package:hypha_wallet/core/ppp_service/service/auth_api.dart';
+import 'package:hypha_wallet/core/ppp_service/service/profile_api.dart';
+
+class PPP {
+  static var instance = PPP();
+
+  Map<String, dynamic>? _config;
+  String? originAppId;
+  AuthApi? _authApi;
+  ProfileApi? _profileApi;
+  String? activeUser;
+
+  // exposting some config params as getters
+  String? get loginContract => _config?['loginContract'];
+  String? get imageTypes => _config?['imageTypes'];
+  int? get maxImageSize => int.parse(_config?['maxImageSize']);
+
+  /// Configures the PPP client
+  /// @param {string} environment to be used test or prod
+  /// @param {string} [originAppId] required for standalone apps, the appId of the app using the PPP client
+  static void configure(String environment, String originAppId) async {
+    instance._config = ppp_config_map[environment];
+    instance.originAppId = originAppId;
+    if (instance._config == null) {
+      throw 'No configuration found for specified environment: $environment';
+    }
+
+    final auth = AmplifyAuthCognito();
+    final api = AmplifyAPI();
+    final storage = AmplifyStorageS3();
+    await Amplify.addPlugins([auth, api, storage]);
+
+    Amplify.configure(instance._config!['AWS']);
+  }
+
+  // we implement change of user later..
+  void checkUser(String activeUser) {
+    if (instance.activeUser != null) {
+      if (instance.activeUser != activeUser) {
+        throw 'Change of users not implemented';
+      }
+      instance.activeUser = activeUser;
+    }
+  }
+
+  /// @returns {AuthApi}
+  static AuthApi authApi(String activeUser) {
+    instance.checkUser(activeUser);
+    instance._authApi ??= AuthApi(activeUser: activeUser, loginContract: instance.loginContract!);
+    return instance._authApi!;
+  }
+
+  /// @returns {ProfileApi}
+  static ProfileApi profileApi(String activeUser) {
+    instance.checkUser(activeUser);
+    instance._profileApi ??= ProfileApi();
+    return instance._profileApi!;
+  }
+
+  // TODO(n13): Put json in file once it's working
+  static const Map<String, dynamic> ppp_config_map = {
```

Comment:
@n13 You've wrote it already, but just to confirm. Config details should be retrieved from environment variables